packages/valory/skills/termination_abci/rounds.py

```diff
@@ -0,0 +1,126 @@
+
+from packages.valory.skills.abstract_round_abci.abci_app_chain import chain
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciApp,
+    AbstractRound,
+    BaseSynchronizedData,
+    BaseTxPayload,
+    CollectSameUntilThresholdRound,
+)
+from packages.valory.skills.termination_abci.payloads import BackgroundPayload
+from packages.valory.skills.transaction_settlement_abci.rounds import (
+    FailedRound,
+    FinishedTransactionSubmissionRound,
+    TransactionSubmissionAbciApp,
+)
+
+
+class Event(Enum):
+    """Defines the (background) round events."""
+
+    TERMINATE = "terminate"
+
+
+class SynchronizedData(BaseSynchronizedData):
+    """
+    Class to represent the synchronized data.
+
+    This data is replicated by the tendermint application.
+    """
+
+    @property
+    def safe_contract_address(self) -> Optional[str]:
+        """Get the safe contract address."""
+        return cast(Optional[str], self.db.get("safe_contract_address", None))
+
+    @property
+    def termination_majority_reached(self) -> bool:
+        """Get termination_majority_reached."""
+        return cast(bool, self.db.get("termination_majority_reached", False))
+
+
+class BackgroundRound(CollectSameUntilThresholdRound):
+    """Defines the background round, which runs concurrently with other rounds."""
+
+    round_id: str = "background_round"
+    allowed_tx_type = BackgroundPayload.transaction_type
+    payload_attribute: str = "background_data"
+    synchronized_data_class = SynchronizedData
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            state = self.synchronized_data.update(
+                synchronized_data_class=self.synchronized_data_class,
+                termination_majority_reached=True,
+                most_voted_tx_hash=self.most_voted_payload,
+            )
+            return state, Event.TERMINATE
+
+        return None
+
+
+class TerminationRound(AbstractRound):
+    """Round to act as the counterpart of the behaviour responsible for terminating the agent."""
+
+    round_id = "termination_round"
+
+    def check_payload(self, payload: BaseTxPayload) -> None:
+        """No logic required here."""
+        pass
+
+    def process_payload(self, payload: BaseTxPayload) -> None:
+        """No logic required here."""
+        pass
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+        """No logic required here."""
+        return None
+
+
+class PostTerminationTxAbciApp(AbciApp):
+    """The abci app running after the multisig transaction has been settled."""
+
+    initial_round_cls = TerminationRound
+    # the following is not needed, it is added to satisfy the round check
+    # the TerminationRound when run it terminates the agent, so nothing can come after it
+    transition_function = {TerminationRound: {Event.TERMINATE: TerminationRound}}
```

Comment:
For this we have degenerate rounds & behaviours. Create one and reference it here?

Reply:
So this satisfies two checks:
1. That the [inital state is in the set of states](https://github.com/valory-xyz/open-autonomy/blob/v0.3.3/packages/valory/skills/abstract_round_abci/base.py#L1741).
2. That the TerminationRound has a [non timeout transition](https://github.com/valory-xyz/open-autonomy/blob/v0.3.3/packages/valory/skills/abstract_round_abci/base.py#L1811).

Reply:
Yes that's what I'm suggesting.